.github/workflows/CD.yml

```diff
@@ -0,0 +1,119 @@
+name: CD
+
+on:
+  push:
+    tags:        
+      - '**' 
+
+jobs:
+  validate-tag:
+    name: Validate tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+      
+      - name: Check no jumps in tags
+        run: |
+          # The format of a tag (version) should be MAJOR.MINOR
+          # If a previous tag is MAJOR.MINOR, valid tags are only MAJOR.(MINOR+1) or (MAJOR+1).0
+          if [[ $(git tag -l | wc -l) != 1 ]]; then # If there is only one tag, no need to check
+            # Find the previous tag
+            prev=$(git tag -l --sort=-v:refname | head -n 2 | tail -n 1)
+            # Split the previous tag into an array of MAJOR and MINOR version components. E.g., for `3.11` it returns the array (3 11)
+            ver_comp=($(tr "." " " <<< $prev))
+            val1="${ver_comp[0]}.$((${ver_comp[1]}+1))" # MAJOR.(MINOR+1)
+            val2="$((${ver_comp[0]}+1)).0" # (MAJOR+1).0
+            if [[ "${{github.ref_name}}" != "$val1" ]] && \
+            [[ "${{github.ref_name}}" != "$val2" ]]; then
+                echo "Invalid tag \"${{github.ref_name}}\" when previous tag is \"$prev\"."
+                exit 1
+            fi
+          fi
```

Comment:
As far as I know we've only imposed strict versioning controls on models and model configurations.

Doing so here, while isn't a bad thing per se, is not consistent with other packages.

So my opinion is that we should remove this. Sorry.

Reply:
Removed in [6d47fe8f75cad1ebaa9b61bb1c585bd9a1229d12](https://github.com/ACCESS-NRI/era5_grib_parallel/pull/3/commits/6d47fe8f75cad1ebaa9b61bb1c585bd9a1229d12)